fix(manifest): ensure manifest author is string, support object with email

hellozzm opened the pull request:
### Overview

Convert author field in manifest to string if it is an object, using the email property. Add test to verify author object is correctly converted to string in the generated manifest. Prevents invalid manifest format in Firefox and other browsers.

### Manual Testing

<!-- Describe how to test your changes to make sure the PR works as intended -->

### Related Issue

<!-- If this PR is related to an issue, please link it here -->

This PR closes #1770

hellozzm commented:
Yes, before making the revisions, I consulted the documentation:
[https://developer.chrome.google.cn/docs/extensions/reference/manifest/author?hl=en](https://developer.chrome.google.cn/docs/extensions/reference/manifest/author?hl=en)

But perhaps this way of handling it can avoid reporting errors on Firefox

Timeraa commented:
Does this also apply to stores like Opera, Edge Addons?

hellozzm commented:
> Does this also apply to stores like Opera, Edge Addons?

<img width="1257" height="490" alt="image" src="https://github.com/user-attachments/assets/554c0f81-8a5c-4d50-9ec0-8dce72121d69" />

As shown in the attached image, the Opera browser extension also encounters the same issue, so the solution used for Firefox can also be applied to Opera. Edge, like Chrome, does not have this problem.

Timeraa commented:
Hey @hellozzm I added some more tests and added a log as from my research as well no one uses the object version of this anymore and only Gecko based browsers use the plain string for display purposes. Could you review what I changed and give your feedback on it?

Timeraa reviewed (approved):
@aklinker1 would like your approval on this change, I don't consider it a breaking change since all browsers don't use it except gecko and we do migration  but even then it's not correctly set I think. So I'd like you to merge this

hellozzm commented:
> Hey @hellozzm I added some more tests and added a log as from my research as well no one uses the object version of this anymore and only Gecko based browsers use the plain string for display purposes. Could you review what I changed and give your feedback on it?

I’ve reviewed your changes—your test cases are much more comprehensive than mine. Overall, everything looks good now.

aklinker1 commented on packages/browser/src/gen/index.d.ts:
Revert this file, it will be overwritten next time the `@types/chrome` package changes. Make the change to that package instead of here.

aklinker1 commented:
See https://github.com/wxt-dev/wxt/issues/1770#issuecomment-3148429940
